syntax = "proto3";

package graph_proto;

message CombineItem {
    double time = 1;
}

message WaitItem {
    double time = 1;
}

message BusItem {
    double time = 1;
    uint32 span_count = 2;
}

message Weight {
    oneof weight {
        CombineItem combine_item = 1;
        WaitItem wait_item = 2;
        BusItem bus_item = 3;
    }
}

message Edge {
    uint64 from = 1;
    uint64 to = 2;
    Weight weight = 3;
}

message IncidenceList {
    repeated uint64 edge_id = 1;
}

message Graph {
    repeated Edge edges = 1;
    repeated IncidenceList incidence_lists = 2;
}